stdlib/public/core/CTypes.swift

```diff
@@ -100,7 +98,19 @@ public typealias CLongDouble = Double
 public typealias CLongDouble = Double
 #endif
 #elseif os(OpenBSD)
+#if arch(x86_64)
+public typealias CLongDouble = Float80
+#else
+#error("CLongDouble needs to be defined for this OpenBSD architecture")
+#endif
+#elseif os(FreeBSD)
+#if arch(x86_64) || arch(i386)
 public typealias CLongDouble = Float80
+#else
+#error("CLongDouble needs to be defined for this FreeBSD architecture")
```

Comment:
Any reason to not define this to `Double`?  The ABI specification clearly indicates that it is FP64 for the other cases.

Reply:
AArch64 and RISCV64 are float128 on FreeBSD too.

Reply:
Right, sorry, that's what I was trying to convey with my comments earlier regarding `__LDBL_MANT_DIG__`.  To elaborate on that, clang shows this built-in macro for the `arm64-freebsd` target:

```
arm64-freebsd: #define __LDBL_MANT_DIG__ 113
```

A 113-bit mantissa points to 128-bit (quad precision) `long double`, which is not the same format as `Double`, which is 64 bits.

Reply:
Ah, right, FP128 is not currently supported in Swift.  Okay, this makes sense.  I suppose that comments to this effect would be helpful for future reference, but 🤷‍♂️.